### PR TITLE
Add mypy environment

### PR DESCRIPTION
**Pull request by wsanchez**
This adds a mypy environment to Tox and run it in Travis.

It is not really doing much or anything at the moment, because this PR doesn't add type annotations to files, and it ignores a few existing errors, with the intention that we can start fixing those errors and adding type annotations in separate PRs.

### CODE REVIEW

File: .travis/mypy [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
```
Comment (moshez) [inline .travis/mypy]: As always I question the wisdom of writing shell scripts rather than Python scripts :)

File: tox.ini [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [tox]
 
 envlist =
-    flake8
+    flake8, mypy
```
Comment (moshez) [inline tox.ini]: Any reason not to add it on its own line?

File: tox.ini [hunk 3]
```diff
@@ -120,6 +120,43 @@ max-complexity = 21
 
 
 ##
+# Mypy linting
+##
+
+[testenv:mypy]
+
+basepython = python3.5
```
Comment (moshez) [inline tox.ini]: Why not 3.6?

Comment (wsanchez) [reply tox.ini]: 3.6 isn't currently in Klein's test matrix.  It should be, but that's another PR, I think.

File: tox.ini (hunk 2 above)
Comment (wsanchez) [reply tox.ini]: No; I just mentally grouped the lint-related environments, and that leaked out here.

File: .travis/mypy (hunk 1 above)
Comment (wsanchez) [reply .travis/mypy]: I hear you, and I've thought of doing this in Python, actually.  This was expedient, and I'm a little bit nervous about using Python in the build pipeline, but my reasons for that probably don't really apply here.  That said, doing so would add very little value here; this is a pretty simple script.

File: tox.ini (hunk 3 above)
Comment (wsanchez) [reply tox.ini]: #218 for tracking.

File: tox.ini (hunk 2 above)
Comment (wsanchez) [reply tox.ini]: split in two

**Pull request merged**